added ambient-link ability sdk

robinlehrmann opened the pull request:
| Q             | A
| ------------- | ---
| License       | MIT

Added ambient.link ability pack.

Nyholm commented:
Thank you for this PR. 

Im not sure why you create a "pack"? Why just not create a recipe for `ambient-link/ability-sdk`?

robinlehrmann commented:
> Thank you for this PR.
> 
> Im not sure why you create a "pack"? Why just not create a recipe for `ambient-link/ability-sdk`?

You are right. I changed the name and removed the pack

Nyholm commented on ambient-link/ability-sdk/1.0/config/packages/ability.yaml:
```diff
@@ -0,0 +1,48 @@
+            $username: '%env(AMBIENT_LINK_MQTT_USERNAME)%'
+            $password: '%env(AMBIENT_LINK_MQTT_PASSWORD)%'
+
+ability:
```
This line and below can be removed.

Nyholm commented on ambient-link/ability-sdk/1.0/config/packages/ability.yaml:
```diff
@@ -0,0 +1,48 @@
+services:
+    AmbientLink\SDK\SmartMonkeyConnection:
+        arguments: ['@ambient_link.connection_settings', '%env(AMBIENT_LINK_USERNAME)%', '%env(AMBIENT_LINK_PASSWORD)%']
```
Dont forget to defined a service for `ambient_link.connection_settings`

Nyholm commented on ambient-link/ability-sdk/1.0/config/packages/ability.yaml:
```diff
@@ -0,0 +1,51 @@
+#                    to: leftRoom
+#                leave_apartment:
+#                   from: leftRoom
+#                    to: notInAnyRoom
```
I suggest removing lines 20 to 51. They dont help the user

Nyholm commented on ambient-link/ability-sdk/1.0/src/AbilityEventSubscriber.php:
```diff
@@ -0,0 +1,37 @@
+use AmbientLink\SDK\Event\EnterRoomEvent;
+use Psr\Log\LoggerInterface;
+
+final class AbilityEventSubscriber implements Ability
```
Is this class really needed? 

I think it can be removed.

Nyholm reviewed (changes requested):
Thank you

Nyholm commented on ambient-link/ability-sdk/1.0/manifest.json:
```diff
@@ -0,0 +1,16 @@
+        "AMBIENT_LINK_PORT": "1883",
+        "AMBIENT_LINK_MQTT_USERNAME":  "",
+        "AMBIENT_LINK_MQTT_PASSWORD": "",
+        "AMBIENT_LINK_BASE_TOPIC": "",
```
This is not used

Nyholm commented on ambient-link/ability-sdk/1.0/manifest.json:
```diff
@@ -0,0 +1,16 @@
+        "AMBIENT_LINK_MQTT_PASSWORD": "",
+        "AMBIENT_LINK_BASE_TOPIC": "",
+        "AMBIENT_LINK_MONKEY_ID":  "",
+        "AMBIENT_LINK_ABILITY_TOKEN": "",
```
This is not used

Nyholm commented on ambient-link/ability-sdk/1.0/manifest.json:
```diff
@@ -0,0 +1,16 @@
+        "AMBIENT_LINK_BASE_TOPIC": "",
+        "AMBIENT_LINK_MONKEY_ID":  "",
+        "AMBIENT_LINK_ABILITY_TOKEN": "",
+        "AMBIENT_LINK_MQTT_USERNAME_AND_PASSWORD": "",
```
This is not used

Nyholm commented on ambient-link/ability-sdk/1.0/manifest.json:
```diff
@@ -0,0 +1,16 @@
+        "AMBIENT_LINK_MONKEY_ID":  "",
+        "AMBIENT_LINK_ABILITY_TOKEN": "",
+        "AMBIENT_LINK_MQTT_USERNAME_AND_PASSWORD": "",
+        "AMBIENT_LINK_MQTT_URL": "${MQTT_USERNAME_AND_PASSWORD}${MQTT_HOST}:1883"
```
This is not used

Nyholm commented on ambient-link/ability-sdk/1.0/config/packages/ability.yaml:
```diff
@@ -0,0 +1,18 @@
+services:
+    AmbientLink\SDK\SmartMonkeyConnection:
+        arguments: ['@amnient.link_connection_settings', '%env(AMBIENT_LINK_USERNAME)%', '%env(AMBIENT_LINK_PASSWORD)%']
```
There is no definition for `AMBIENT_LINK_USERNAME` or `AMBIENT_LINK_PASSWORD`

Nyholm reviewed (changes requested):
Thank you. 

Can you make sure that you define all environment variables? Also, dont define more environment variables than you are using.

Nyholm reviewed (approved):
Thank you.
